chainntnfs/bitcoindnotify/bitcoind.go

```diff
@@ -525,46 +526,102 @@ func (b *BitcoindNotifier) confDetailsManually(confRequest chainntnfs.ConfReques
+		if err != nil {
+			return nil, chainntnfs.TxNotFoundManually,
+				fmt.Errorf("unable to receive the block "+
+					"promise %v", err)
+		}
+
+		batchRequest[height] = sendReq
+	}
+
+	// Sending the bulk request using the updated batchClient.
+	b.chainConn.SendAsyncQueue()
+
+	blockHashAddresses := make([]*chainhash.Hash, 0, len(batchRequest))
+
+	for height, response := range batchRequest {
 		// Ensure we haven't been requested to shut down before
-		// processing the next height.
+		// processing the next height response.
 		select {
 		case <-b.quit:
 			return nil, chainntnfs.TxNotFoundManually,
 				chainntnfs.ErrChainNotifierShuttingDown
 		default:
 		}
 
-		blockHash, err := b.chainConn.GetBlockHash(int64(height))
+		// First, we'll fetch the block for the current height.
+		blockHash, err := response.Receive()
 		if err != nil {
 			return nil, chainntnfs.TxNotFoundManually,
-				fmt.Errorf("unable to get hash from block "+
-					"with height %d", height)
+				fmt.Errorf("unable to retrieve hash for block "+
+					"with height %d: %v", height, err)
 		}
 
-		block, err := b.GetBlock(blockHash)
-		if err != nil {
+		blockHashAddresses = append(blockHashAddresses, blockHash)
+	}
+
+	// Now we fetch blocks in interval of 15 to avoid out of memory
+	// errors in case of fetching 1000 blocks from GetBlocksBatch().
+	intervals := len(blockHashAddresses) / 15
+
+	for i := 0; i <= intervals; i++ {
+		// Ensure we haven't been requested to shut down before
+		// processing next set of blocks.
+		select {
+		case <-b.quit:
 			return nil, chainntnfs.TxNotFoundManually,
-				fmt.Errorf("unable to get block with hash "+
-					"%v: %v", blockHash, err)
+				chainntnfs.ErrChainNotifierShuttingDown
+		default:
 		}
 
-		// For every transaction in the block, check which one matches
-		// our request. If we find one that does, we can dispatch its
-		// confirmation details.
-		for txIndex, tx := range block.Transactions {
-			if !confRequest.MatchesTx(tx) {
-				continue
+		var (
+			blocks []*wire.MsgBlock
+			err    error
+		)
+		strt, end := i*15, (i*15)+15
+		if i == intervals {
+			if len(blockHashAddresses)%15 == 0 {
+				break
 			}
+			blocks, err = b.chainConn.GetBlocksBatch(
+				blockHashAddresses[strt:],
+			)
+		} else {
+			blocks, err = b.chainConn.GetBlocksBatch(
+				blockHashAddresses[strt:end],
+			)
+		}
+		if err != nil {
+			return nil, chainntnfs.TxNotFoundManually, err
+		}
 
-			return &chainntnfs.TxConfirmation{
-				Tx:          tx,
-				BlockHash:   blockHash,
-				BlockHeight: height,
-				TxIndex:     uint32(txIndex),
-			}, chainntnfs.TxFoundManually, nil
+		height := int(currentHeight) - strt
+		for j := range blocks {
+			// For every transaction in the block, check which one
+			// matches our request. If we find one that does, we can
+			// dispatch its confirmation details.
+			for txIndex, tx := range blocks[j].Transactions {
+				if !confRequest.MatchesTx(tx) {
+					height--
```

Comment:
This seems to be incorrect to me as here we're continuing the inner for loop over the transactions. The height should only be decreased if none of the txes match. So maybe a cleaner way is to return if `confRequest.MatchesTx(tx)` otherwise decrease `height` outside before continuing the loop over the blocks.